src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -11,16 +11,16 @@ internal static partial class GlobalizationMode
         // split from GlobalizationMode so the whole class can be trimmed when Invariant=true.
         private static partial class Settings
         {
-            internal static readonly bool PredefinedCulturesOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY");
             internal static bool Invariant { get; } = GetInvariantSwitchValue();
+            internal static readonly bool PredefinedCulturesOnly = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```

Comment:
Why are Invariant and PredefinedCulturesOnly not symmetric in their design?  e.g. I was expecting to see:
```C#
internal static bool Invariant { get; } = GetInvariantSwitchValue();
internal static bool PredefinedCulturesOnly { get; } = GetPredefinedCulturesOnlySwitchValue();
```

Reply:
Because we decided to not introduce a new config switch with the new Invariant behavior. @jkotas suggested to use the `PredefinedCulturesOnly` switch which make a lot of sense. Now the default Invariant behavior is to throw for any non-Invariant culture. For that we have to have `PredefinedCulturesOnly` be true by default for Invariant. and by false by default for non-Invariant mode.

Reply:
I was asking a style question. GetPredefinedCulturesOnlySwitchValue() would return `AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);`, exactly what you're already initializing it to.

Reply:
@stephentoub I am going to merge this change soon. please let me know if you have any concern.

Reply:
I just don't see why the shape of these is different.  But it's not a blocker, it's just inconsistent.

Reply:
Sorry my reply crossed your reply. This is how it is done to help the linker for trimming purpose I guess. look at @marek-safar change https://github.com/dotnet/runtime/commit/bc27d49a2ed20029fd1d3630e21cd2b54ef75d54 which introduced that.

Reply:
I think this done this way to allow the linker to trim `GetInvariantSwitchValue` when Invariant mode is one. Having `GetPredefinedCulturesOnlySwitchValue` will require more work from the linker to trim it too. @eerhardt or @marek-safar should have definitive answer here.

Reply:
> I think this done this way to allow the linker to trim GetInvariantSwitchValue when Invariant mode is one.

I don't believe this is the case. I think it is inconsistent because this code has been refactored multiple times, and no one has made it consistent.

> Having GetPredefinedCulturesOnlySwitchValue will require more work from the linker to trim it too. 

I don't believe this is true either. Alternatively, the opposite approach could be taken to make the code look consistent:

```C#
internal static bool Invariant { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
internal static bool PredefinedCulturesOnly { get; } = AppContextConfigHelper.GetBooleanConfig("System.Globalization.PredefinedCulturesOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_CULTURES_ONLY", Invariant);
```

Reply:
I'll try to submit a PR to make this consistent.